### ISSUE TITLE
Conflict with `display_derive`

### ISSUE THREAD

**Issue by ErichDonGubler** [Peternator7/strum#30]
The `Display` procedural macro unfortunately conflicts with @withoutboats' [`display_derive`](https://github.com/withoutboats/display_derive) procedural macro. In some code I'm working with on a personal project, the following will compile when `#[macro_use]`ing only `display_derive`, but not in conjunction with `strum_derive`:

```rust
#[derive(Clone, Debug, Deserialize, Display, Eq, Hash, PartialEq, Serialize)]
#[display(fmt = "{}", _0)]
pub struct MessageId(pub String);
```

Both of these crates are great additions to the ecosystem, in my opinion. It's a shame that we don't have macros as first-class items yet and that I have to choose between them. Perhaps disabling `Display` with a feature flag would be best for now?

EDIT: It's also possible to use the right crate's definition by making the desired derive be imported LAST...but that's a hacky workaround.

**Comment by lo48576** [comment]
Can we use more detailed name, such as `EnumDisplay`?
I'm not sure whether it should be recommended rather than generic name `Display`, or whether it should always be available.
But at least when `Display` is disabled, such alternative name may be useful because it lets strum to avoid degrading its feature.

**Comment by Peternator7** [comment]
@ErichDonGubler  Thanks for the PR and being patient while I find time to look at it :) 

@lo48576  Are you suggesting we change the name across the board or just when this feature flag is flipped? I wish I knew the roadmap for macros a better. To entirely future-proof this issue, we could just have a flag `PrependStrum` that for all the derives puts strum in front of the attribute: `StrumToString` , `StrumDisplay`, etc. It's a little ugly for sure, but it would make it easy to avoid collisions across the whole crate.

**Comment by lo48576** [comment]
I meant renaming only `Display` to some other name (and maybe also `As{Static,Ref}str`).
It's because `Display` and `AsRefStr` is too generic and other crates may implement them for non-enum types.

In contrast, `EnumString` and something like that is cleary for enum, and if other crates wanted to use that name, I think the feature is not provided by two crates for the same code.
For example, if both strum and other crate (call it alt_strum) provides `derive(EnumDisplay)`, then users should not use them at the same time.
(It can happen, but it is not desirable to use two crates for same purpose, I think.
For such case, strum or alt_strum should cover all custom derives required by the user.)

To make strum future-proof (not only avoiding conflicts), using strum's own namespace (such as `#[derive(Strum)] #[strum(derive(Display))]`) may be good.

**Comment by ErichDonGubler** [comment]
My PR wholesale *disables* `Display`, but renaming to indicate that the proc-macro comes from `StrumDisplay` seems like an equally viable solution. I would personally prefer something like that to what I submitted -- simply omitting it was the simplest to implement at the time. :)

**Comment by Peternator7** [comment]
@ErichDonGubler , I'm going to close this. Let me know if you disagree.

**Comment by ErichDonGubler** [comment]
@Peternator7: Haven't heard a response from this, just wanted to ping you.

**Comment by Peternator7** [comment]
Hey @ErichDonGubler , thanks for following up on this. It had slipped my mind. That's interesting official distros are so old. Let me figure out what the earliest version of rustc compiles strum. I don't plan on having extensive long term support, but at least making sure new features don't break back compat is a good start. I'll get a travis build set up for that version.

**Comment by Peternator7** [comment]
Following up on this. The earliest version of rust right now that can compile strum is 1.26 because that's when impl trait was stabilized so I made a note on the README.

It's should be possible to rename macros with some conditional feature flags like we discussed above. I'm planning on creating one flag per custom derive because the overhead for adding a feature is so small, and typically the conflict only exists with a single name so consumers won't need to refactor their entire library, only colliding names. 

Thanks for your patience on this. I've been busy with other things lately.

```
#[cfg_attr(not(feature = "verbose-enumstring-name"), proc_macro_derive(EnumString, attributes(strum)))]
#[cfg_attr(feature = "verbose-enumstring-name", proc_macro_derive(StrumEnumString, attributes(strum)))]
pub fn from_string(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
    let ast = syn::parse(input).unwrap();

    let toks = from_string::from_string_inner(&ast);
    debug_print_generated(&ast, &toks);
    toks.into()
}
```

**Comment by Peternator7** [comment]
*Housekeeping*

This was implemented in `0.15.0` so I believe this issue is okay to close now.
